oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/celliterator/CellIterator.java

```diff
@@ -526,74 +545,106 @@ public IterateAllEntry(
    *         intervals.
    */
   public Stream<IterateAllEntry> iterateByContribution(GridOSHEntity cell) {
-    OSHDBTimestampInterval timeInterval = new OSHDBTimestampInterval(timestamps);
-
+    var cellBoundingBox = XYGrid.getBoundingBox(new CellId(
+        cell.getLevel(),
+        cell.getId()
+    ), true);
     final boolean allFullyInside;
     if (isBoundByPolygon) {
-      // if cell is fully inside bounding box/polygon we can skip all entity-based inclusion checks
-      OSHDBBoundingBox cellBoundingBox = XYGrid.getBoundingBox(new CellId(
-          cell.getLevel(),
-          cell.getId()
-      ), true);
       if (bboxOutsidePolygon.test(cellBoundingBox)) {
         return Stream.empty();
       }
       allFullyInside = bboxInPolygon.test(cellBoundingBox);
     } else {
       allFullyInside = false;
     }
+    return iterateByContribution(cell.getEntities(), allFullyInside);
+  }
+
+  /**
+   * Helper method to easily iterate over all entity modifications in a cell that match a given
+   * condition/filter.
+   *
+   * @param cellData the entities to iterate through
+   * @param allFullyInside
+   *
+   * @return a stream of matching filtered OSMEntities with their clipped Geometries and timestamp
+   *         intervals.
+   */
+  public Stream<IterateAllEntry> iterateByContribution(Iterable<? extends OSHEntity> cellData,
```

Comment:
actually not, if we make this private then we don't needed it at all. 
But you are right, instead of passing the `allFullyInside` we could pass the bbox instead and do the `allFullyInside` calculation in the method with the Iterable argument.

Reply:
hmm, AFAICS the method is only used internally by `iterateByContribution(GridOSHEntity cell)` at the moment. Did you intend that end users would call this method directly? If yes, what's the benefit over just using the simpler `iterateByContribution(GridOSHEntity cell)` version where one doesn't have to worry about providing a correct value for `allFullyInside` (or alternatively the cell's `bbox`)?

Reply:
I reverted my comment, so that it does not pass the bbox but the flag instead.
It is sometimes usefull to use this util function to iterate over an other source than just GridOSHEntities.
For example, if you have an OSHWay and want to have for all nodes they contribution history with tags.
```
cellIterator.iterateByContributions(oshWay.getNodes(), true);
```
Admittedly this is very artificial. Just to show the point.
Most often I needed this method if you wanted to test e.g. contributions for a single or a list of OSHEntities for the ease without to create a grid from them. But I doubt that this is a good reason to make it public.
The standard user hardly will ever use the cellIterator directly. So I guess it will be ok if this is public, but we can make it private or revert it so its original stat as you like.